app/src/main/java/chat/rocket/android/wallet/create/ui/CreateWalletFragment.kt

```diff
@@ -0,0 +1,143 @@
+package chat.rocket.android.wallet.create.ui
+
+import android.content.Context
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.support.v7.view.ActionMode
+import android.view.*
+import android.widget.Toast
+import android.util.Log
+import chat.rocket.android.R
+import chat.rocket.android.R.menu.password
+import chat.rocket.android.util.extensions.*
+import chat.rocket.android.util.extensions.inflate
+import chat.rocket.android.wallet.create.presentation.CreateWalletPresenter
+import chat.rocket.android.wallet.create.presentation.CreateWalletView
+import dagger.android.support.AndroidSupportInjection
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
+import io.reactivex.rxkotlin.Observables
+import javax.inject.Inject
+import kotlinx.android.synthetic.main.fragment_create_wallet.*
+import timber.log.Timber
+
+class CreateWalletFragment:  Fragment(), CreateWalletView, android.support.v7.view.ActionMode.Callback {
+    @Inject lateinit var presenter: CreateWalletPresenter
+    private var actionMode: ActionMode? = null
+    private val disposables = CompositeDisposable()
+
+    companion object {
+        fun newInstance() = CreateWalletFragment()
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        AndroidSupportInjection.inject(this)
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? = container?.inflate(R.layout.fragment_create_wallet)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        disposables.add(listenToChanges())
+    }
+
+    override fun onDestroyView() {
+        disposables.clear()
+        super.onDestroyView()
+    }
+
+    override fun onActionItemClicked(mode: ActionMode, menuItem: MenuItem): Boolean {
+
```

Comment:
maybe we should require a password of certain length / complexity

Reply:
I'll update the required character count (8+ characters), but NIST recommends NOT requiring arbitrary symbols/numbers. Did you have something else in mind for increasing complexity? https://www.passwordping.com/surprising-new-password-guidelines-nist/